sys/include/net/loramac.h

```diff
@@ -0,0 +1,715 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_loramac LoRaMAC
+ * @ingroup     net
+ * @brief       LoRaMAC definitions
+ *
+ * @{
+ *
+ * @file
+ * @brief       LoRaMAC header definitions
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef NET_LORAMAC_H
+#define NET_LORAMAC_H
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    LoRaMAC default parameters
+ * @{
+ */
+/**
+ * @brief   Default device EUI
+ *
+ *          8 bytes key, required for join procedure
+ */
+#ifndef LORAMAC_DEV_EUI_DEFAULT
+#define LORAMAC_DEV_EUI_DEFAULT        { 0x00, 0x00, 0x00, 0x00, \
+                                         0x00, 0x00, 0x00, 0x00 }
+#endif
+
+/**
+ * @brief   Default application EUI
+ *
+ *          8 bytes key, required for join procedure
+ */
+#ifndef LORAMAC_APP_EUI_DEFAULT
+#define LORAMAC_APP_EUI_DEFAULT        { 0x00, 0x00, 0x00, 0x00, \
+                                         0x00, 0x00, 0x00, 0x00 }
+#endif
+
+/**
+ * @brief   Default application key
+ *
+ *          16 bytes key, required for join procedure
+ */
+#ifndef LORAMAC_APP_KEY_DEFAULT
+#define LORAMAC_APP_KEY_DEFAULT        { 0x00, 0x00, 0x00, 0x00, \
+                                         0x00, 0x00, 0x00, 0x00, \
+                                         0x00, 0x00, 0x00, 0x00, \
+                                         0x00, 0x00, 0x00, 0x00 }
```

Comment:
Those keys should definitely not be 0....

Reply:
Since this is something that is set by the user retrieved from the LoRaWAN backend, I think these list can be kept like this by default.
The other problem is that some keys are not needed depending on the activation mode (ABP: requires all keys, OTAA only requires dev eui, app eui and app key).

Are you convinced ?

Reply:
Dito